packages/airnode-node/src/api/index.ts

```diff
@@ -0,0 +1,233 @@
+        apiCredentials,
+        metadata: null,
+      };
+    }
+    case 'regular': {
+      const { airnodeAddress, requesterAddress, sponsorAddress, sponsorWalletAddress, endpointId, id, chainId } =
+        aggregatedApiCall;
+      const chain = config.chains.find((c) => c.id === chainId)!;
+
+      return {
+        endpointName,
+        parameters: sanitizedParameters,
+        ois,
+        apiCredentials,
+        metadata: {
+          airnodeAddress: airnodeAddress,
+          requesterAddress: requesterAddress,
+          sponsorAddress: sponsorAddress,
+          sponsorWalletAddress: sponsorWalletAddress,
+          endpointId: endpointId,
+          requestId: id,
+          chainId: chainId,
+          chainType: chain.type,
+          airnodeRrpAddress: chain.contracts.AirnodeRrp,
+        },
+      };
+    }
+  }
+}
+
+async function signResponseMessage(requestId: string, responseValue: string, config: Config) {
+  const masterHDNode = getMasterHDNode(config);
+  const airnodeWallet = ethers.Wallet.fromMnemonic(masterHDNode.mnemonic!.phrase);
+
+  return await airnodeWallet.signMessage(
+    ethers.utils.arrayify(
+      ethers.utils.keccak256(ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId, responseValue || '0x']))
+    )
+  );
+}
+
+export interface CallApiPayload {
+  readonly config: Config;
+  readonly aggregatedApiCall: AggregatedApiCall;
+}
+
+function verifySponsorWallet(
+  config: Config,
+  aggregatedApiCall: RegularAggregatedApiCall
+): LogsData<ApiCallErrorResponse> | null {
+  const { sponsorAddress, sponsorWalletAddress, id } = aggregatedApiCall;
+  const hdNode = getMasterHDNode(config);
+  if (isValidSponsorWallet(hdNode, sponsorAddress, sponsorWalletAddress)) return null;
+
+  // TODO: Abstract this to a logging utils file
+  const message = `${RequestErrorMessage.SponsorWalletInvalid}, Request ID:${id}`;
+  const log = logger.pend('ERROR', message);
+  return [
+    [log],
+    {
+      success: false,
+      errorMessage: message,
+    },
+  ];
+}
+
+function verifyRequestId(
+  config_: Config,
+  aggregatedApiCall: RegularAggregatedApiCall
+): LogsData<ApiCallErrorResponse> | null {
+  if (isValidRequestId(aggregatedApiCall)) return null;
+
+  const message = `${RequestErrorMessage.RequestIdInvalid}. Request ID:${aggregatedApiCall.id}`;
+  const log = logger.pend('ERROR', message);
+  return [
+    [log],
+    {
+      success: false,
+      errorMessage: message,
+    },
+  ];
+}
+
+export function verifyTemplateId(
+  config_: Config,
+  aggregatedApiCall: RegularAggregatedApiCall
+): LogsData<ApiCallErrorResponse> | null {
+  const { templateId, template, id } = aggregatedApiCall;
+  if (!templateId) {
+    return null;
+  }
+
+  if (!template) {
+    const message = `Ignoring Request:${id} as the template could not be found for verification`;
+    const log = logger.pend('ERROR', message);
+    return [
+      [log],
+      {
+        success: false,
+        errorMessage: message,
```

Comment:
Could save a few lines by moving everything to one line?

```ts
return [[log], { success: false, errorMessage: message }];
```

1 line vs 7 lines